src/Xamarin.Android.Tools.Aidl/CSharpCodeGenerator.cs

```diff
@@ -613,7 +630,15 @@ string GetWriteStatements (TypeName type, string parcel, string arg, string parc
 					return String.Format ("{1}.WriteStrongBinder (((({0} != null)) ? ({0}.AsBinder ()) : (null)));", arg, parcel);
 			}
 		}
-		
+
+		string GetWriteOutStatements (TypeName type, string parcel, string arg)
+		{
+			if (type.ArrayDimension > 0) {
```

Comment:
…ditto here: is `int[][]` possible?

A quick gander at the [AIDL documentation](https://developer.android.com/guide/components/aidl) suggests it *isn't* possible.  However, it *does* suggest that `List`:

> must be one of the supported data types in this list or one of the other AIDL-generated interfaces or parcelables you've declared

which implies that a `List<List<List<String>>>` may be viable…and I have no idea what we'd do with that.  (Wouldn't impact `type.ArrayDimension`, though!)

Reply:
The way I understand the AIDL dokumentation is that multidimensional arrays like `int[][]` are not supported. At the moment the `CSharpCodeGenerator` or maybe the `AidlParser` just ignores higher dimensions and int[][] is treated like int[]. However throwing some NotSupportedException might not be wrong.

For `List` I think - but not 100% sure - you can put any supported object in. And one object can be a List which can contain other Lists and so on. I don't know why (maybe a bug) but the Android Studio does not support parametrization like `List<List<List<String>>>>`, just `List<String>`. So you can use `List` and put Lists in but `List<List>>` does not work!?

The CSharpCodeGenerator might not cover all cases for Lists correct and should definetely be unit tested also in this perspect.